api/base/parsers.py

```diff
@@ -208,3 +210,27 @@ def flatten_relationships(self, relationships):
         if ret.get('target_type') and ret.get('id'):
             return {related_resource: ret['id']}
         return ret
+
+
+class HMACSignedParser(JSONParser):
+
+    def parse(self, stream, media_type=None, parser_context=None):
+        """
+        Parses the incoming bytestream as JSON. Validates the 'signature' in the payload then returns the resulting data.
+        """
+        data = super(HMACSignedParser, self).parse(stream, media_type=media_type, parser_context=parser_context)
+
+        try:
```

Comment:
Minor: it would be useful to know that this code was copied from `must_be_signed`. Maybe add a code comment above `must_be_signed` to remove it once Waterbutler is sending requests to the v2 endpoints.